HungryPhilosophers/src/Philosopher.java

```diff
@@ -0,0 +1,60 @@
+import java.util.Random;
+
+class Philosopher implements Runnable {
+		String threadName;
+		int n;
+		Fork leftFork, rightFork;
+		
+		public Philosopher(String threadName, int n, Fork leftFork, Fork rightFork) {
+			this.threadName = threadName;
+			this.n = n;
+			this.leftFork=leftFork;
+			this.rightFork=rightFork;
+		}
+		
+        public boolean accessingFork(Fork leftFork, Fork rightFork) {
+			boolean isLeftForkFree = false;
+			boolean isRightForkFree = false;
+            try {
+            	isLeftForkFree = leftFork.lock.tryLock();
+            	isRightForkFree = rightFork.lock.tryLock();
```

Comment:
I think you shouldn't be greedy and not even try to get grep of right fork if left one is unavailable.